ops/model.py

```diff
@@ -1324,7 +1364,228 @@ def list_files(self, path: str, *, pattern: str = None,
             itself: If path refers to a directory, return information about the
                 directory itself, rather than its contents.
         """
-        return self._pebble.list_files(path, pattern=pattern, itself=itself)
+        return self._pebble.list_files(str(path), pattern=pattern, itself=itself)
+
+    def push_path(self,
```

Comment:
This was @niemeyer's suggestion.  The idea was to capture the semantics of being able to push either a dir (recursively) or a file.  But I see how it could be read as unhelpfully vague.  Maybe - `push_anything`? I think I lean toward `push_recursive` as a preference, but I'm okay either way.

Reply:
Fair enough, if it's already been discussed, I'm okay with it. `push_path` is not terrible. :-)